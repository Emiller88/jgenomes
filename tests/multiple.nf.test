nextflow_pipeline {

    name "Build multiple references"
    script "../main.nf"
    tag "pipeline"

    test("-profile test --tools bwamem1,createsequencedictionary,faidx,intervals --input assets/test/default_multiple.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "${projectDir}/assets/test/default_multiple.yml"
                outdir = "$outputDir"
                tools  = 'bwamem1,createsequencedictionary,faidx,intervals'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("-profile test --tools tabix --input assets/test/pipelines/sarek_s3_muliple_glob.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "${projectDir}/assets/test/pipelines/sarek_s3_muliple_glob.yml"
                outdir = "$outputDir"
                tools  = 'tabix'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}', 'Homo_sapiens/GATK/*/Annotation/GATK_BUNDLE/*.vcf.gz'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore', ignore: ['Homo_sapiens/GATK/*/Annotation/GATK_BUNDLE/*.vcf.gz'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }
}
