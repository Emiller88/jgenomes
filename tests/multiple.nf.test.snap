{
    "Run with profile test for multiple with globs on s3 | --input assets/test/pipelines/sarek_s3_muliple_glob.yml": {
        "content": [
            5,
            {
                "TABIX_TABIX": {
                    "tabix": 1.2
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "Homo_sapiens",
                "Homo_sapiens/GATK",
                "Homo_sapiens/GATK/GRCh37",
                "Homo_sapiens/GATK/GRCh37/Annotation",
                "Homo_sapiens/GATK/GRCh37/Annotation/GATK_BUNDLE",
                "Homo_sapiens/GATK/GRCh37/Annotation/GATK_BUNDLE/1000G_phase1.indels.b37.vcf.gz.tbi",
                "Homo_sapiens/GATK/GRCh37/Annotation/GATK_BUNDLE/Mills_and_1000G_gold_standard.indels.b37.vcf.gz.tbi",
                "Homo_sapiens/GATK/GRCh38",
                "Homo_sapiens/GATK/GRCh38/Annotation",
                "Homo_sapiens/GATK/GRCh38/Annotation/GATK_BUNDLE",
                "Homo_sapiens/GATK/GRCh38/Annotation/GATK_BUNDLE/Homo_sapiens_assembly38.known_indels.vcf.gz.tbi",
                "Homo_sapiens/GATK/GRCh38/Annotation/GATK_BUNDLE/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi",
                "index.json",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml"
            ],
            [
                "1000G_phase1.indels.b37.vcf.gz.tbi:md5,aa36bfab96e05ff18f9f58925a41306a",
                "Mills_and_1000G_gold_standard.indels.b37.vcf.gz.tbi:md5,d3f818e36a0d4fa2bcdfbbca47b5e325",
                "Homo_sapiens_assembly38.known_indels.vcf.gz.tbi:md5,5d0aad81fa364a93ba4dff95bb6fb766",
                "Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi:md5,c04253adf2c4192475110557086d5ac1",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-02T12:09:53.42046409"
    },
    "Run with profile test for multiple | --input assets/test/default_multiple.yml": {
        "content": [
            7,
            {
                "BUILD_INTERVALS": {
                    "gawk": "5.3.0"
                },
                "BWAMEM1_INDEX": {
                    "bwa": "0.7.18-r1243-dirty"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "Homo_sapiens",
                "Homo_sapiens/nf-core",
                "Homo_sapiens/nf-core/references",
                "Homo_sapiens/nf-core/references/GRCh38_chr21",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation/intervals",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation/intervals/GRCh38_chr21.bed",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex/version0.7.18",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex/version0.7.18/GRCh38_chr21.amb",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex/version0.7.18/GRCh38_chr21.ann",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex/version0.7.18/GRCh38_chr21.bwt",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex/version0.7.18/GRCh38_chr21.pac",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/BWAIndex/version0.7.18/GRCh38_chr21.sa",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/WholeGenomeFasta",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/WholeGenomeFasta/GRCh38_chr21.fa.fai",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/intervals",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Annotation/intervals/testdata.GRCh38_chr22.bed",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.amb",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.ann",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.bwt",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.pac",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/BWAIndex/version0.7.18/genome.sa",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/WholeGenomeFasta",
                "Homo_sapiens/nf-core/references/testdata.GRCh38_chr22/Sequence/WholeGenomeFasta/genome.fasta.fai",
                "index.json",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml"
            ],
            [
                "GRCh38_chr21.bed:md5,472d213cfcde96565699779d5bfc0e32",
                "GRCh38_chr21.amb:md5,e59d7ca1d4e9593989a13925bfc15c07",
                "GRCh38_chr21.ann:md5,4d42ab1762829012e9139cacbeea7615",
                "GRCh38_chr21.bwt:md5,a5015025a0ae077c011833bd73651794",
                "GRCh38_chr21.pac:md5,8c08bed90eb472d0d35451e378e399f3",
                "GRCh38_chr21.sa:md5,9abb4ddd6fb4eab96b79f48540882329",
                "GRCh38_chr21.fa.fai:md5,6ea51e1c1231da5e49212ea9f493d703",
                "testdata.GRCh38_chr22.bed:md5,87a15eb9c2ff20ccd5cd8735a28708f7",
                "genome.amb:md5,1891c1de381b3a96d4e72f590fde20c1",
                "genome.ann:md5,2df4aa2d7580639fa0fcdbcad5e2e969",
                "genome.bwt:md5,815eded87e4cb6b0f1daab5c4d6e30af",
                "genome.pac:md5,8569fbdb2c98c6fb16dfa73d8eacb070",
                "genome.sa:md5,e7cff62b919448a3a3d0fe4aaf427594",
                "genome.fasta.fai:md5,3520cd30e1b100e55f578db9c855f685",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-10T13:44:39.079626708"
    }
}
